components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa.Protocol/src/Services/SubscriptionServices.cs

```diff
@@ -1249,22 +1272,32 @@ private class MonitoredItemWrapper {
                 }
                 while (superType != null);
 
-                var propertyNames = new List<QualifiedName>();
+                var fieldNames = new List<QualifiedName>();
                 foreach (var node in nodes) {
-                    foreach (var reference in node.ReferenceTable) {
-                        if (reference.ReferenceTypeId == ReferenceTypeIds.HasProperty) {
-                            propertyNames.Add(nodeCache.FetchNode(reference.TargetId).BrowseName);
-                        }
-                    }
+                    ParseFields(nodeCache, fieldNames, node);
                 }
+                fieldNames = fieldNames
+                    .OrderBy(x => x.Name).ToList();
```

Comment:
Sanity check: If you reorder it... will the indexing work in GetFieldDisplayName?

Reply:
Yes, the FieldNames list is built later, based on the actual SelectClause, so yes - two separate things.